module/Application/src/Application/Authentication/DomainBased/DomainBasedAuthentication.php

```diff
@@ -0,0 +1,64 @@
+<?php 
+
+namespace Application\Authentication\DomainBased;
+
+use Zend\Authentication\Adapter\AdapterInterface;
+use Zend\Authentication\Result;
+use Zend\Http\Request;
+use Application\Service\UserService;
+use Application\Entity\User;
+
+
+class DomainBasedAuthentication implements AdapterInterface{
```

Comment:
Rivedi il nome della classe, in questo modo:
1. Metti la classe nel modulo ZFX\Authentication
2. Chiama la classe DomainAdapter